data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3SelectObjectWorker.java

```diff
@@ -4,22 +4,12 @@
  */
```

Comment:
I was getting an NPE in some of the testing from a bug in the OpenSearch retry strategy. If the entry in the map does not exist, an NPE is thrown when trying to unbox `null` into the primitive `int`.

https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java#L205

This is unrelated to this PR but does affect the overall S3 select functionality. I was able to solve it locally by adding this above line 205:
```
if (!retryCountMap.containsKey(bulkRequestForRetry)) {
    retryCountMap.put(bulkRequestForRetry, 1);
}
```

Could you include that fix as well? @kkondaka might have some input on whether or not the above snippet is the correct solution